### PR TITLE
remove deprecated Cluster.Address field (replaced by Cluster.Status.address)

### PR DESCRIPTION
**Pull request by xrstf**
**What this PR does / why we need it**:
KKP 2.21 moved and migrated the address, it's now possible to get rid of the deprecated field alltogether.

/kind cleanup

**Does this PR introduce a user-facing change? Then add your Release Note here**:
```release-note
BREAKING CHANGE: The `address` field in the Cluster CRD was deprecated in KKP 2.21 and removed in this release. Use `status.address` instead. Existing clusters were migrated automatically by the seed-controller-manager in release 2.21.
```

### CODE REVIEW

File: pkg/resources/prometheus/configmap.go [hunk 1]
```diff
@@ -392,7 +392,7 @@ scrape_configs:
   scheme: https
   tls_config:
 {{ .ApiserverTLSConfig | indent 4 }}
-  
+
```
Comment (xrstf) [inline pkg/resources/prometheus/configmap.go:395]: FYI, this tiny little change where I removed the trailing newlines caused the major formatting changes in the YAML fixtures. :grin:

**Review by ahmedwaleedmalik — APPROVED** [review]
/approve

**Pull request merged**